sdk/samples/iot/iot_sample_common.c

```diff
@@ -46,27 +47,27 @@
 //
 //#define USE_WEB_SOCKET // Comment to use MQTT without WebSockets.
 #ifdef USE_WEB_SOCKET
-const az_span mqtt_url_prefix = AZ_SPAN_LITERAL_FROM_STR("wss://");
+az_span const mqtt_url_prefix = AZ_SPAN_LITERAL_FROM_STR("wss://");

```

Comment:
ew I don't like this but if our sdk guideline overlords say so I'll get used to it.

ps: actually a lot of the guidelines seem to have it on the left for declaration. 
https://azure.github.io/azure-sdk/clang_design.html#globals
@ahsonkhan are we to go through with this mismatch?

Reply:
I don't think we ever discussed the ordering of const for local variables, or even that you should use them.

I have no opinion one way or the other. For local variables, I am fine with const on the left, or not even using it, unless you really care/benefit from knowing it won't change as you write the code.

To me, its an implementation detail/preference that may not necessarily belong in the guidelines since it shouldn't impact the end user.

Reply:
@barcharcraz - where are const for local variables supposed to go? And do we need to update the guidelines to be consistent?

Reply:
I just realized @danewalton that you are referring to public globals, not locals.

RHS const is more consistent.

As an aside, we should follow the global naming guidelines (capitalized and `AZ_...`).

Reply:
const should be on the left side. Putting it after the makes pointer look weird. 

`Object const * const var = NULL;`

vs

`const Object* const var = NULL;`

Reply:
lots here.  sounds like we are doing const RHS always, regardless of local, global, ptr, non ptr, etc.  si?

I'm seeing internal vs. private terminology in the link dane shared. Does this sound accurate:
Public - means we support the function/global variable as part of the public API.
Internal - means we don't support the function/global variable as part of the public API.

If a global is static, then it must ALSO be internal. It should use _az_BLAH_BLAH_BLAH.
If a global is internal, that does NOT mean it is has to be static.  It should still use _az_BLAH_BLAH_BLAH
If a global is public, then it is not internal, meaning it is also not static.  It should use AZ_BLAH_BLAH_BLAH.

Reply:
side: I would prefer if we were consistent with RHS everywhere for const, otherwise it gets challenging to remember when the rule applies and when it doesn't.

Reply:
> const should be on the left side. Putting it after the makes pointers look weird.

@ericwol-msft - we discussed this in our last Friday meeting. We use `const* const` in all such places in core.

Reply:
> I'm seeing internal vs. private terminology in the link dane shared. Does this sound accurate:
Public - means we support the function/global variable as part of the public API.
Internal - means we don't support the function/global variable as part of the public API.

> If a global is static, then it must ALSO be internal. It should use _az_BLAH_BLAH_BLAH.
If a global is internal, that does NOT mean it is has to be static. It should still use _az_BLAH_BLAH_BLAH
If a global is public, then it is not internal, meaning it is also not static. It should use AZ_BLAH_BLAH_BLAH.

Yes. That's right.

> ... and what do we do about the `g`?  is it g_az_BLAH, and G_AZ_BLAH?  or g_AZ_BLAH?

Nothing, we should update the guidelines to remove that. We use such prefixes in C++ (`g_`, `s_`, `c_`, `m_`).

Reply:
On second thought, we are not in the AZ API at all with these samples.  Note, iot_smple_common functions and variables do not begin with _az_iot_sample.  They begin with iot_sample.  

With that format, it would seem the globals should be named g_iot_sample_MQTT_URL_PREFIX

Reply:
> Yes. That's right. Although...
> 
> > If a global is static, then it must ALSO be internal.
> 
> I don't think that's true (and does it matter?). Where did you read that?

I didn't read that, I was inferring it.  So it could be static (private to the file), but public with respect to the API?  I guess, why would be do that?  No developer should be using it if it is private to the file...?

Reply:
The question is meaningless when it comes to public static globals in header files:
https://stackoverflow.com/q/11967502

Reply:
that's a great point, since a header is the public API.  I agree.  Adding to list topics I emailed out...

Reply:
> As an aside, we should follow the global naming guidelines (capitalized and `AZ_...`).

Realized these actually should be static.  Scoped to the file, but not true globals.  The guidelines actually don't say anything about static global variables that are scoped to file.  They do mention static functions though, which these are like.  In that case, no prefix is needed at all.

Adding this topic to the ongoing list of discrepancies.